### PR TITLE
chore(eslint): enable prefer-nullish-coalescing rule

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Enable typescript eslint [_prefer-nullish-coalescing_](https://typescript-eslint.io/rules/prefer-nullish-coalescing) rule.

> The `??` nullish coalescing runtime operator allows providing a default value when dealing with `null` or `undefined`. Because the nullish coalescing operator only coalesces when the original value is `null` or `undefined`, it is much safer than relying upon logical OR operator chaining `||`, which coalesces on any falsy value.

I've configure it to ignore conditional cases and `string` and `boolean` primitives.

## Justify why they are needed

This is part of a series of PRs that includes new/updated rules added by typescript-eslint v6. The idea is to use those PRs so we discuss as a team if we want to enable a particular rule or not.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2916** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2916" target="_blank">Graphite</a> 
    * **PR #2917** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2917" target="_blank">Graphite</a> 
      * **PR #2918** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2918" target="_blank">Graphite</a> 
        * **PR #2919** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2919" target="_blank">Graphite</a> 
          * **PR #2920** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2920" target="_blank">Graphite</a> 
            * **PR #2923** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923?utm_source=stack-comment).

File: apps/store/src/services/storyblok/Storyblok.helpers.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const fetchStory = async <StoryData extends ISbStoryData>(
 ): Promise<StoryData> => {
   let cv: number | undefined
   if (params.version === 'published') {
-    cv = (STORYBLOK_CACHE_VERSION && parseInt(STORYBLOK_CACHE_VERSION)) || undefined
+    cv = STORYBLOK_CACHE_VERSION ? parseInt(STORYBLOK_CACHE_VERSION) : undefined
```
Comment (guilhermespopolin) [inline apps/store/src/services/storyblok/Storyblok.helpers.ts:46]: This is probably incorret.

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by guilhermespopolin** [comment]
@guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923).

**Comment by guilhermespopolin** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923) couldn't merge this pull request because a downstack PR #2916 failed to merge.

**Comment by guilhermespopolin** [comment]
@guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923).

**Comment by guilhermespopolin** [comment]
[Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2923) rebased this pull request as part of a merge.

**Pull request merged**